# Install BigDFT code or client in Google Colab, locally or into your google drive folder
This notebook will install the code in your google drive folder, which will be useful to excute the other tutorials of the school.

You should provide authorisation to access your google drive after the execution of the first cell.

The files of the school will go in the `bigdft-school` directory. Such folder can me moved to your local disk or deleted once you are done.

The entire installation procedure should take about one minute.

In the forthcoming lessons, some other packages will be also installed in this folder.

In [1]:
!wget https://gitlab.com/luigigenovese/bigdft-school/-/raw/main/packaging/install.py &> /dev/null
import install
install.full_suite() # Kernel will restart after this, it is normal. Count about 3 minutes

The notebook should have restarted at present. You may want to remove the runtime warning at the bottom of the page.

The install module will have to be imported again to set the notebook ready for execution.

To see if everything looks OK you can execute this cell to see if an energy value is output at the end:

In [ ]:
import install
install.set_ready() #this resets the basic environment variables and puts us in the correct directory, (only needed for the full suite installation)

In [ ]:
# Test that we can import everything
import futile
import BigDFT

# Test A Full Calculation with PyBigDFT
from BigDFT.Systems import System
from BigDFT.Fragments import Fragment
from BigDFT.Atoms import Atom

at = Atom({"He": [0, 0, 0]})
frag = Fragment([at])
sys = System({"FRA:0": frag})

from BigDFT.Calculators import SystemCalculator
code = SystemCalculator(skip=True) #this skip option would not run the code if the result is present.

from BigDFT.Inputfiles import Inputfile
inp = Inputfile()
inp.set_xc("PBE")
inp.set_hgrid(0.4)

log = code.run(sys=sys, input=inp)
print(log.energy)



Initialize a Calculator with OMP_NUM_THREADS=1 and command /content/bigdft-school/install/bin/bigdft
Creating the yaml input file "./input.yaml"
Executing command:  /content/bigdft-school/install/bin/bigdft -s Yes
-2.8893983558799454


Such energy value should be `-2.8893983558799454` (Hartree)

We can now present another possible installation of the code, which is performed via only the bigdft client. This installation is lighter as it only involves the python pakages employed by PyBigDFT for pre- and post- processing, and it will be enough for most of the tutorials of the school.


In [ ]:
install.client(locally=True)

# Exercise 1
Rerun the notebook with the client instead to see if the energy is the same even without the BigDFT executable.

Tips: restart the runtime without erasing the files, and run the installation command.

# Exercise 2

Install the bigdft client on your google drive folder to continue with the school trainings.

To trigger this installation you have to authorize the usage of your google drive folder. This would prevent the installation to be performed evey time.

Tips: it is just enough to remove the ``locally=True`` option.


In [ ]:
# write your answer here

In [3]:
#we flush the google drive at the end of the process (this can take a long time. You can install locally the client if it takes too long)
install.close_drive()